### DISCUSSION TITLE
How to auto-populate the gallery

### DISCUSSION THREAD

**Discussion by boatloader** [nextapps-de/spotlight#90] [Q&A]
Hi all,

I've found this project a few days ago and it seems to fulfill my needs for a nice gallery / media-showroom framework.
But I did not find out, how I can auto-populate the gallery.

In my project, my picture library is constantly growing and it would be nice to have a "hot folder", which is watched by spotlight.

As soon as I add/modify/delete media there, how can I achieve, that I don't have to name every media object in code?

In the best case, spotlight scans my hot folder, and puts "albums" on the main page, which are represented in the file system by sub-folders in the hot folder.

As soon as I move/modify/rename any data, the website content is updated accordingly.

Is there a way to achieve this?

For context: I'm somewhat new to web-programming, but I'm familiar with python, C, C++ etc., so maybe there are ways to transfer my knowledge to web-programming ;-)

Cheers

**Comment by RolandReyer** [comment]
I’m sure you can run a process on your own server to watch a folder and rebuild a gallery when necessary.
But the gallery can’t do that by itself, that would be a security issue.
I’ve built a gallery for my photos using using data from an asset management (iMatch) and some Python scripts. The server is my NAS.

**Reply by boatloader** [reply]
Hi Roland,

thank you for the hints and informations!

Where does the security problem result from / what is the problem, if auto-population is active?

Best regards

**Comment by RolandReyer** [comment]
If a *website* could access your filesystem (to check if there are new files) then ANYONE could have access to your files.
That's why this not possible at all.

To see what's on your HD there must be a process running *on your computer* that provides the information, like lists of file names, content etc.
If you build a HTML file for a gallery then the image files have to be accessible for the webserver.

I have used the "Dominate" Python library to create the gallery pages.

**Reply by boatloader** [reply]
Thanks a lot!
I will have a look into this, sounds promising!

**Comment by boatloader** [comment]
One additional question:
In the README.md the API of spotlight is described.
In one section it says 

```
Basic Usage (API)

Also you can programmatically use Spotlight via the library API. This way does not require any dependant HTML elements (e.g. the classname "spotlight").

Define a gallery group as follows:

var gallery = [
    { src: "cat1.jpg" },
    { src: "cat2.jpg" },
    { src: "cat3.jpg" }
];
```

Is this API meant to be called from within another java script, or can it be called from e.g. python?

I'm asking, since this may also be a way, to tell spotlight about the image files that it has to display, or am I wrong?

If so, might this be a way to scan e.g. every day for new items in a folder with e.g. a python script and if new images were found, pass the array of images to spotlight.

Will something along those lines work?
